static/js/components/EducationTab.js

```diff
@@ -2,50 +2,95 @@ import React from 'react';
-      { value: 'b', label: "Bachelor's degree", enabled: true },
-      { value: 'm', label: "Master's or professional degree", enabled: false },
-      { value: 'p', label: "Doctorate", enabled: false }
+      {value: HIGH_SCHOOL, label: "High school"},
+      {value: ASSOCIATE, label: 'Associate degree'},
+      {value: BACHELORS, label: "Bachelor's degree"},
+      {value: MASTERS, label: "Master's or professional degree"},
+      {value: DOCTORATE, label: "Doctorate"}
     ];
-    this.educationLevelLabels = {
-      'hs': "High school",
-      'a': 'Associate degree',
-      'b': "Bachelor's degree",
-      'm': "Master's or professional degree",
-      'p': "Doctorate"
-    };
 
-    this.saveAndContinue = saveAndContinue.bind(this, '/dashboard');
+    this.educationLevelLabels = {};
+    this.educationLevelOptions.forEach(level => {
+      this.educationLevelLabels[level.value] = level.label;
+    });
+    this.saveAndContinue = saveAndContinue.bind(this, '/profile/professional', EducationTab.validation);
     this.openNewEducationForm = this.openNewEducationForm.bind(this);
     this.handleCloseDialog = this.handleCloseDialog.bind(this);
 
   }
-  openNewEducationForm(level, index){
-    const { dispatch, profile, updateProfile } = this.props;
+  static propTypes ={
+    educationLevels: React.PropTypes.object,
+    educationDialog: React.PropTypes.object,
+    profile: React.PropTypes.object,
+    saveProfile: React.PropTypes.func,
+    updateProfile: React.PropTypes.func,
+    errors: React.PropTypes.object,
+    dispatch: React.PropTypes.func.isRequired
+  };
+
+  static defaultProps = {
+    requiredFields: [],
+    validationMessages: {
+      'degree_name': "'Degree level",
+      'graduation_date': 'Graduation Date',
+      'field_of_study': 'Field of study',
+      'online_degree': 'Online Degree',
+      'school_name': 'School name',
+      'school_city': 'City',
+      'school_state_or_territory': 'State',
+      'school_country': 'Country'
+    }
+  };
+
+  static validation(profile, requiredFields) {
+    const keys = [
+      'degree_name',
+      'graduation_date',
+      'field_of_study',
+      'online_degree',
+      'school_name',
+      'school_city',
+      'school_state_or_territory',
+      'school_country'
+    ];
+    let nestedFields = (index) => {
+      let keySet = (key) => ['education', index, key];
+      return keys.map(key => keySet(key));
+    };
+    return requiredFields.concat(
+      ...profile.education.map((v, i) => nestedFields(i))
+    );
+  }
+
+  openNewEducationForm(level, index) {
+    const {dispatch, profile, updateProfile} = this.props;
     let new_index = index;
     if (index === null){
       new_index = profile['education'].length;
     }
     /* add empty education */
     let clone = Object.assign({}, profile);
-    clone['education'].push(this.addNewEducation(level));
+    clone['education'].push(generateNewEducation(level));
```

Comment:
You can't use `push` here since it alters the list and you've only made a shallow copy. Can you rewrite using `concat` instead?